### DISCUSSION TITLE
Cannot log in - Requires flow_name or flow_token

### DISCUSSION THREAD

**Discussion by alarno** [mahrtayyab/tweety#195] [Q&A]
Hello,

I've been using `tweety` for a while now to interact with Twitter's API. However, as of today, I'm encountering an issue where the login process fails, requesting `flow_name` or `flow_token`. This problem started suddenly and has halted my ability to authenticate.

Here is the error message I am receiving:
Error signing in as [username]: [357] Either flow_name or flow_token need to be defined

Here is the code snippet I am using to sign in:

```python
from tweety import Twitter
import os

def sign_in(username, password):
    try:
        # Remove the previous session if it exists
        if os.path.exists("./session.tw_session"):
            os.remove("./session.tw_session")

        # Create a new Twitter instance
        app = Twitter("session")

        # Attempt to log in
        app.start(username, password)

        # Save the session for future use
        app.session.save()

        print(f"Logged in as {username}")
        return app
    except Exception as e:
        print(f"Error signing in as {username}: {e}")
        return None

# Credentials
username = "xxxxx"
password = "xxxx"

# Attempt to log in
app = sign_in(username, password)

**Comment by mahrtayyab** [comment]
Please do use `sign_in` or `start` method of `Twitter` class , it will automatically verifies the saved session (if any), if that session was expired it will automatically login and save the session. Logging in again and again is not something encouraged by Twitter and you most likely have to perform Captcha in order to login, I am investigating further.

**Comment by mahrtayyab** [comment]
also make sure you are on latest version

**Comment by alarno** [comment]
Hi,

I have updated the `tweety-ns` library to the latest version using the following command:

```sh
pip install --upgrade tweety-ns

# Importing the necessary library
from tweety import Tweet

# Attempting to sign in as user xxxxx
try:
    tweet = Tweet.sign_in(username='xxxxxx', password='your_password_here')
except Exception as e:
    print(f"Error signing in as xxxx: {e}")

The error message I receive is:

Error signing in as xxxxxx: [357] Either flow_name or flow_token need to be defined

It seems that even after updating the library, the error persists. Could you please advise on how to resolve this issue?

Thank you!

**Comment by mahrtayyab** [comment]
Do update to latest version and try
